zeebe/qa/update-tests/src/test/java/io/camunda/zeebe/test/VersionCompatibilityMatrix.java

```diff
@@ -254,10 +254,19 @@ private Stream<Ref> fetchTags() {
                         HttpRequest.newBuilder().GET().uri(endpoint).build(),
                         BodyHandlers.ofByteArray()));
 
+        final var statusCode = response.statusCode();
+        if (statusCode < 200 || statusCode >= 300) {
+          final var body = new String(response.body());
+          throw new RuntimeException(
+              String.format(
+                  "Failed to fetch tags from GitHub API. Status: %d, Endpoint: %s, Response: %s",
+                  statusCode, endpoint, body));
+        }
+
         final var refs = new ObjectMapper().readValue(response.body(), Ref[].class);
         return Stream.of(refs);
       } catch (final Exception e) {
```

Comment:
🙃 Could we narrow down the catch to the specific exception classes now? Otherwise we'll wrap the new `RuntimeException` thrown above again in another `RuntimeException`.